Fails to build on 32-bit architectures (e.g., armv7, i386, powerpc)

Regressed by 06b91cfdc21e. Originally, [found](https://pkg-status.freebsd.org/beefy17/data/main-i386-default/p7df5d2371cbe_s3f960a05ba0/logs/errors/vkmark-2017.08.36_1.log) [downstream](https://github.com/freebsd/freebsd-ports/commit/4868165d658c). Curiously, 06b91cfdc21e is part of `v1.3.302` in Vulkan-Hpp but part of `v1.4.303` in Vulkan-Headers.
```c++
$ echo 'int main() {}' | clang++ -m32 -x c++ -include vulkan/vulkan.hpp -</dev/null
In file included from <built-in>:1:
In file included from /usr/include/vulkan/vulkan.hpp:8864:
/usr/include/vulkan/vulkan_structs.hpp:1241:90: error: invalid operands to binary expression ('const vk::AccelerationStructureKHR' and 'const vk::AccelerationStructureKHR')
 1241 |       return ( sType == rhs.sType ) && ( pNext == rhs.pNext ) && ( accelerationStructure == rhs.accelerationStructure ) &&
      |                                                                    ~~~~~~~~~~~~~~~~~~~~~ ^  ~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/vulkan/vulkan.hpp:203:8: note: candidate template ignored: could not match 'const ArrayWrapper1D<char, N>' against 'const vk::AccelerationStructureKHR'
  203 |   bool operator==( ArrayWrapper1D<char, N> const & lhs, ArrayWrapper1D<char, M> const & rhs ) VULKAN_HPP_NOEXCEPT
      |        ^
/usr/include/vulkan/vulkan.hpp:248:8: note: candidate template ignored: could not match 'const ArrayWrapper1D<char, N>' against 'const vk::AccelerationStructureKHR'
  248 |   bool operator==( std::string const & lhs, ArrayWrapper1D<char, N> const & rhs ) VULKAN_HPP_NOEXCEPT
      |        ^
/usr/include/vulkan/vulkan_enums.hpp:166:29: note: candidate template ignored: could not match 'const Flags<BitType>' against 'const vk::AccelerationStructureKHR'
  166 |   VULKAN_HPP_CONSTEXPR bool operator==( BitType bit, Flags<BitType> const & flags ) VULKAN_HPP_NOEXCEPT
      |                             ^
In file included from <built-in>:1:
In file included from /usr/include/vulkan/vulkan.hpp:8864:
/usr/include/vulkan/vulkan_structs.hpp:1242:40: error: invalid operands to binary expression ('const vk::AccelerationStructureNV' and 'const vk::AccelerationStructureNV')
 1242 |              ( accelerationStructureNV == rhs.accelerationStructureNV );
      |                ~~~~~~~~~~~~~~~~~~~~~~~ ^  ~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/vulkan/vulkan.hpp:203:8: note: candidate template ignored: could not match 'const ArrayWrapper1D<char, N>' against 'const vk::AccelerationStructureNV'
  203 |   bool operator==( ArrayWrapper1D<char, N> const & lhs, ArrayWrapper1D<char, M> const & rhs ) VULKAN_HPP_NOEXCEPT
      |        ^
/usr/include/vulkan/vulkan.hpp:248:8: note: candidate template ignored: could not match 'const ArrayWrapper1D<char, N>' against 'const vk::AccelerationStructureNV'
  248 |   bool operator==( std::string const & lhs, ArrayWrapper1D<char, N> const & rhs ) VULKAN_HPP_NOEXCEPT
      |        ^
/usr/include/vulkan/vulkan_enums.hpp:166:29: note: candidate template ignored: could not match 'const Flags<BitType>' against 'const vk::AccelerationStructureNV'
  166 |   VULKAN_HPP_CONSTEXPR bool operator==( BitType bit, Flags<BitType> const & flags ) VULKAN_HPP_NOEXCEPT
      |                             ^
In file included from <built-in>:1:
In file included from /usr/include/vulkan/vulkan.hpp:8864:
/usr/include/vulkan/vulkan_structs.hpp:1389:113: error: invalid operands to binary expression ('const vk::Buffer' and 'const vk::Buffer')
 1389 |       return ( sType == rhs.sType ) && ( pNext == rhs.pNext ) && ( createFlags == rhs.createFlags ) && ( buffer == rhs.buffer ) && ( offset == rhs.offset ) &&
      |                                                                                                          ~~~~~~ ^  ~~~~~~~~~~
/usr/include/vulkan/vulkan.hpp:203:8: note: candidate template ignored: could not match 'const ArrayWrapper1D<char, N>' against 'const vk::Buffer'
  203 |   bool operator==( ArrayWrapper1D<char, N> const & lhs, ArrayWrapper1D<char, M> const & rhs ) VULKAN_HPP_NOEXCEPT
      |        ^
/usr/include/vulkan/vulkan.hpp:248:8: note: candidate template ignored: could not match 'const ArrayWrapper1D<char, N>' against 'const vk::Buffer'
  248 |   bool operator==( std::string const & lhs, ArrayWrapper1D<char, N> const & rhs ) VULKAN_HPP_NOEXCEPT
      |        ^
/usr/include/vulkan/vulkan_enums.hpp:166:29: note: candidate template ignored: could not match 'const Flags<BitType>' against 'const vk::Buffer'
  166 |   VULKAN_HPP_CONSTEXPR bool operator==( BitType bit, Flags<BitType> const & flags ) VULKAN_HPP_NOEXCEPT
      |                             ^
In file included from <built-in>:1:
In file included from /usr/include/vulkan/vulkan.hpp:8864:
/usr/include/vulkan/vulkan_structs.hpp:1595:79: error: invalid operands to binary expression ('const vk::Buffer' and 'const vk::Buffer')
 1595 |       return ( sType == rhs.sType ) && ( pNext == rhs.pNext ) && ( vertexData == rhs.vertexData ) && ( vertexOffset == rhs.vertexOffset ) &&
      |                                                                    ~~~~~~~~~~ ^  ~~~~~~~~~~~~~~
/usr/include/vulkan/vulkan.hpp:203:8: note: candidate template ignored: could not match 'const ArrayWrapper1D<char, N>' against 'const vk::Buffer'
  203 |   bool operator==( ArrayWrapper1D<char, N> const & lhs, ArrayWrapper1D<char, M> const & rhs ) VULKAN_HPP_NOEXCEPT
      |        ^
/usr/include/vulkan/vulkan.hpp:248:8: note: candidate template ignored: could not match 'const ArrayWrapper1D<char, N>' against 'const vk::Buffer'
  248 |   bool operator==( std::string const & lhs, ArrayWrapper1D<char, N> const & rhs ) VULKAN_HPP_NOEXCEPT
      |        ^
/usr/include/vulkan/vulkan_enums.hpp:166:29: note: candidate template ignored: could not match 'const Flags<BitType>' against 'const vk::Buffer'
  166 |   VULKAN_HPP_CONSTEXPR bool operator==( BitType bit, Flags<BitType> const & flags ) VULKAN_HPP_NOEXCEPT
      |                             ^
In file included from <built-in>:1:
In file included from /usr/include/vulkan/vulkan.hpp:8864:
/usr/include/vulkan/vulkan_structs.hpp:1597:26: error: invalid operands to binary expression ('const vk::Buffer' and 'const vk::Buffer')
 1597 |              ( indexData == rhs.indexData ) && ( indexOffset == rhs.indexOffset ) && ( indexCount == rhs.indexCount ) && ( indexType == rhs.indexType ) &&
      |                ~~~~~~~~~ ^  ~~~~~~~~~~~~~
/usr/include/vulkan/vulkan.hpp:203:8: note: candidate template ignored: could not match 'const ArrayWrapper1D<char, N>' against 'const vk::Buffer'
  203 |   bool operator==( ArrayWrapper1D<char, N> const & lhs, ArrayWrapper1D<char, M> const & rhs ) VULKAN_HPP_NOEXCEPT
      |        ^
/usr/include/vulkan/vulkan.hpp:248:8: note: candidate template ignored: could not match 'const ArrayWrapper1D<char, N>' against 'const vk::Buffer'
  248 |   bool operator==( std::string const & lhs, ArrayWrapper1D<char, N> const & rhs ) VULKAN_HPP_NOEXCEPT
      |        ^
/usr/include/vulkan/vulkan_enums.hpp:166:29: note: candidate template ignored: could not match 'const Flags<BitType>' against 'const vk::Buffer'
  166 |   VULKAN_HPP_CONSTEXPR bool operator==( BitType bit, Flags<BitType> const & flags ) VULKAN_HPP_NOEXCEPT
      |                             ^
In file included from <built-in>:1:
In file included from /usr/include/vulkan/vulkan.hpp:8864:
/usr/include/vulkan/vulkan_structs.hpp:1598:30: error: invalid operands to binary expression ('const vk::Buffer' and 'const vk::Buffer')
 1598 |              ( transformData == rhs.transformData ) && ( transformOffset == rhs.transformOffset );
      |                ~~~~~~~~~~~~~ ^  ~~~~~~~~~~~~~~~~~
/usr/include/vulkan/vulkan.hpp:203:8: note: candidate template ignored: could not match 'const ArrayWrapper1D<char, N>' against 'const vk::Buffer'
  203 |   bool operator==( ArrayWrapper1D<char, N> const & lhs, ArrayWrapper1D<char, M> const & rhs ) VULKAN_HPP_NOEXCEPT
      |        ^
/usr/include/vulkan/vulkan.hpp:248:8: note: candidate template ignored: could not match 'const ArrayWrapper1D<char, N>' against 'const vk::Buffer'
  248 |   bool operator==( std::string const & lhs, ArrayWrapper1D<char, N> const & rhs ) VULKAN_HPP_NOEXCEPT
      |        ^
/usr/include/vulkan/vulkan_enums.hpp:166:29: note: candidate template ignored: could not match 'const Flags<BitType>' against 'const vk::Buffer'
  166 |   VULKAN_HPP_CONSTEXPR bool operator==( BitType bit, Flags<BitType> const & flags ) VULKAN_HPP_NOEXCEPT
      |                             ^
In file included from <built-in>:1:
In file included from /usr/include/vulkan/vulkan.hpp:8864:
/usr/include/vulkan/vulkan_structs.hpp:1731:77: error: invalid operands to binary expression ('const vk::Buffer' and 'const vk::Buffer')
 1731 |       return ( sType == rhs.sType ) && ( pNext == rhs.pNext ) && ( aabbData == rhs.aabbData ) && ( numAABBs == rhs.numAABBs ) && ( stride == rhs.stride ) &&
      |                                                                    ~~~~~~~~ ^  ~~~~~~~~~~~~
/usr/include/vulkan/vulkan.hpp:203:8: note: candidate template ignored: could not match 'const ArrayWrapper1D<char, N>' against 'const vk::Buffer'
  203 |   bool operator==( ArrayWrapper1D<char, N> const & lhs, ArrayWrapper1D<char, M> const & rhs ) VULKAN_HPP_NOEXCEPT
      |        ^
/usr/include/vulkan/vulkan.hpp:248:8: note: candidate template ignored: could not match 'const ArrayWrapper1D<char, N>' against 'const vk::Buffer'
  248 |   bool operator==( std::string const & lhs, ArrayWrapper1D<char, N> const & rhs ) VULKAN_HPP_NOEXCEPT
      |        ^
/usr/include/vulkan/vulkan_enums.hpp:166:29: note: candidate template ignored: could not match 'const Flags<BitType>' against 'const vk::Buffer'
  166 |   VULKAN_HPP_CONSTEXPR bool operator==( BitType bit, Flags<BitType> const & flags ) VULKAN_HPP_NOEXCEPT
      |                             ^
In file included from <built-in>:1:
In file included from /usr/include/vulkan/vulkan.hpp:8864:
/usr/include/vulkan/vulkan_structs.hpp:2308:90: error: invalid operands to binary expression ('const vk::AccelerationStructureKHR' and 'const vk::AccelerationStructureKHR')
 2308 |       return ( sType == rhs.sType ) && ( pNext == rhs.pNext ) && ( accelerationStructure == rhs.accelerationStructure );
      |                                                                    ~~~~~~~~~~~~~~~~~~~~~ ^  ~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/vulkan/vulkan.hpp:203:8: note: candidate template ignored: could not match 'const ArrayWrapper1D<char, N>' against 'const vk::AccelerationStructureKHR'
  203 |   bool operator==( ArrayWrapper1D<char, N> const & lhs, ArrayWrapper1D<char, M> const & rhs ) VULKAN_HPP_NOEXCEPT
      |        ^
/usr/include/vulkan/vulkan.hpp:248:8: note: candidate template ignored: could not match 'const ArrayWrapper1D<char, N>' against 'const vk::AccelerationStructureKHR'
  248 |   bool operator==( std::string const & lhs, ArrayWrapper1D<char, N> const & rhs ) VULKAN_HPP_NOEXCEPT
      |        ^
/usr/include/vulkan/vulkan_enums.hpp:166:29: note: candidate template ignored: could not match 'const Flags<BitType>' against 'const vk::AccelerationStructureKHR'
  166 |   VULKAN_HPP_CONSTEXPR bool operator==( BitType bit, Flags<BitType> const & flags ) VULKAN_HPP_NOEXCEPT
      |                             ^
In file included from <built-in>:1:
In file included from /usr/include/vulkan/vulkan.hpp:8864:
/usr/include/vulkan/vulkan_structs.hpp:2854:114: error: invalid operands to binary expression ('const vk::AccelerationStructureNV' and 'const vk::AccelerationStructureNV')
 2854 |       return ( sType == rhs.sType ) && ( pNext == rhs.pNext ) && ( type == rhs.type ) && ( accelerationStructure == rhs.accelerationStructure );
      |                                                                                            ~~~~~~~~~~~~~~~~~~~~~ ^  ~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/vulkan/vulkan.hpp:203:8: note: candidate template ignored: could not match 'const ArrayWrapper1D<char, N>' against 'const vk::AccelerationStructureNV'
  203 |   bool operator==( ArrayWrapper1D<char, N> const & lhs, ArrayWrapper1D<char, M> const & rhs ) VULKAN_HPP_NOEXCEPT
      |        ^
/usr/include/vulkan/vulkan.hpp:248:8: note: candidate template ignored: could not match 'const ArrayWrapper1D<char, N>' against 'const vk::AccelerationStructureNV'
  248 |   bool operator==( std::string const & lhs, ArrayWrapper1D<char, N> const & rhs ) VULKAN_HPP_NOEXCEPT
      |        ^
/usr/include/vulkan/vulkan_enums.hpp:166:29: note: candidate template ignored: could not match 'const Flags<BitType>' against 'const vk::AccelerationStructureNV'
  166 |   VULKAN_HPP_CONSTEXPR bool operator==( BitType bit, Flags<BitType> const & flags ) VULKAN_HPP_NOEXCEPT
      |                             ^
In file included from <built-in>:1:
In file included from /usr/include/vulkan/vulkan.hpp:8864:
/usr/include/vulkan/vulkan_structs.hpp:4353:78: error: invalid operands to binary expression ('const vk::SwapchainKHR' and 'const vk::SwapchainKHR')
 4353 |       return ( sType == rhs.sType ) && ( pNext == rhs.pNext ) && ( swapchain == rhs.swapchain ) && ( timeout == rhs.timeout ) &&
      |                                                                    ~~~~~~~~~ ^  ~~~~~~~~~~~~~
/usr/include/vulkan/vulkan.hpp:203:8: note: candidate template ignored: could not match 'const ArrayWrapper1D<char, N>' against 'const vk::SwapchainKHR'
  203 |   bool operator==( ArrayWrapper1D<char, N> const & lhs, ArrayWrapper1D<char, M> const & rhs ) VULKAN_HPP_NOEXCEPT
      |        ^
/usr/include/vulkan/vulkan.hpp:248:8: note: candidate template ignored: could not match 'const ArrayWrapper1D<char, N>' against 'const vk::SwapchainKHR'
  248 |   bool operator==( std::string const & lhs, ArrayWrapper1D<char, N> const & rhs ) VULKAN_HPP_NOEXCEPT
      |        ^
/usr/include/vulkan/vulkan_enums.hpp:166:29: note: candidate template ignored: could not match 'const Flags<BitType>' against 'const vk::SwapchainKHR'
  166 |   VULKAN_HPP_CONSTEXPR bool operator==( BitType bit, Flags<BitType> const & flags ) VULKAN_HPP_NOEXCEPT
      |                             ^
In file included from <built-in>:1:
In file included from /usr/include/vulkan/vulkan.hpp:8864:
/usr/include/vulkan/vulkan_structs.hpp:4354:26: error: invalid operands to binary expression ('const vk::Semaphore' and 'const vk::Semaphore')
 4354 |              ( semaphore == rhs.semaphore ) && ( fence == rhs.fence ) && ( deviceMask == rhs.deviceMask );
      |                ~~~~~~~~~ ^  ~~~~~~~~~~~~~
/usr/include/vulkan/vulkan.hpp:203:8: note: candidate template ignored: could not match 'const ArrayWrapper1D<char, N>' against 'const vk::Semaphore'
  203 |   bool operator==( ArrayWrapper1D<char, N> const & lhs, ArrayWrapper1D<char, M> const & rhs ) VULKAN_HPP_NOEXCEPT
      |        ^
/usr/include/vulkan/vulkan.hpp:248:8: note: candidate template ignored: could not match 'const ArrayWrapper1D<char, N>' against 'const vk::Semaphore'
  248 |   bool operator==( std::string const & lhs, ArrayWrapper1D<char, N> const & rhs ) VULKAN_HPP_NOEXCEPT
      |        ^
/usr/include/vulkan/vulkan_enums.hpp:166:29: note: candidate template ignored: could not match 'const Flags<BitType>' against 'const vk::Semaphore'
  166 |   VULKAN_HPP_CONSTEXPR bool operator==( BitType bit, Flags<BitType> const & flags ) VULKAN_HPP_NOEXCEPT
      |                             ^
In file included from <built-in>:1:
In file included from /usr/include/vulkan/vulkan.hpp:8864:
/usr/include/vulkan/vulkan_structs.hpp:4354:56: error: invalid operands to binary expression ('const vk::Fence' and 'const vk::Fence')
 4354 |              ( semaphore == rhs.semaphore ) && ( fence == rhs.fence ) && ( deviceMask == rhs.deviceMask );
      |                                                  ~~~~~ ^  ~~~~~~~~~
/usr/include/vulkan/vulkan.hpp:203:8: note: candidate template ignored: could not match 'const ArrayWrapper1D<char, N>' against 'const vk::Fence'
  203 |   bool operator==( ArrayWrapper1D<char, N> const & lhs, ArrayWrapper1D<char, M> const & rhs ) VULKAN_HPP_NOEXCEPT
      |        ^
/usr/include/vulkan/vulkan.hpp:248:8: note: candidate template ignored: could not match 'const ArrayWrapper1D<char, N>' against 'const vk::Fence'
  248 |   bool operator==( std::string const & lhs, ArrayWrapper1D<char, N> const & rhs ) VULKAN_HPP_NOEXCEPT
      |        ^
/usr/include/vulkan/vulkan_enums.hpp:166:29: note: candidate template ignored: could not match 'const Flags<BitType>' against 'const vk::Fence'
  166 |   VULKAN_HPP_CONSTEXPR bool operator==( BitType bit, Flags<BitType> const & flags ) VULKAN_HPP_NOEXCEPT
      |                             ^
In file included from <built-in>:1:
In file included from /usr/include/vulkan/vulkan.hpp:8864:
/usr/include/vulkan/vulkan_structs.hpp:7449:90: error: invalid operands to binary expression ('const vk::AccelerationStructureNV' and 'const vk::AccelerationStructureNV')
 7449 |       return ( sType == rhs.sType ) && ( pNext == rhs.pNext ) && ( accelerationStructure == rhs.accelerationStructure ) && ( memory == rhs.memory ) &&
      |                                                                    ~~~~~~~~~~~~~~~~~~~~~ ^  ~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/vulkan/vulkan.hpp:203:8: note: candidate template ignored: could not match 'const ArrayWrapper1D<char, N>' against 'const vk::AccelerationStructureNV'
  203 |   bool operator==( ArrayWrapper1D<char, N> const & lhs, ArrayWrapper1D<char, M> const & rhs ) VULKAN_HPP_NOEXCEPT
      |        ^
/usr/include/vulkan/vulkan.hpp:248:8: note: candidate template ignored: could not match 'const ArrayWrapper1D<char, N>' against 'const vk::AccelerationStructureNV'
  248 |   bool operator==( std::string const & lhs, ArrayWrapper1D<char, N> const & rhs ) VULKAN_HPP_NOEXCEPT
      |        ^
/usr/include/vulkan/vulkan_enums.hpp:166:29: note: candidate template ignored: could not match 'const Flags<BitType>' against 'const vk::AccelerationStructureNV'
  166 |   VULKAN_HPP_CONSTEXPR bool operator==( BitType bit, Flags<BitType> const & flags ) VULKAN_HPP_NOEXCEPT
      |                             ^
In file included from <built-in>:1:
In file included from /usr/include/vulkan/vulkan.hpp:8864:
/usr/include/vulkan/vulkan_structs.hpp:7449:133: error: invalid operands to binary expression ('const vk::DeviceMemory' and 'const vk::DeviceMemory')
 7449 |       return ( sType == rhs.sType ) && ( pNext == rhs.pNext ) && ( accelerationStructure == rhs.accelerationStructure ) && ( memory == rhs.memory ) &&
      |                                                                                                                              ~~~~~~ ^  ~~~~~~~~~~
/usr/include/vulkan/vulkan.hpp:203:8: note: candidate template ignored: could not match 'const ArrayWrapper1D<char, N>' against 'const vk::DeviceMemory'
  203 |   bool operator==( ArrayWrapper1D<char, N> const & lhs, ArrayWrapper1D<char, M> const & rhs ) VULKAN_HPP_NOEXCEPT
      |        ^
/usr/include/vulkan/vulkan.hpp:248:8: note: candidate template ignored: could not match 'const ArrayWrapper1D<char, N>' against 'const vk::DeviceMemory'
  248 |   bool operator==( std::string const & lhs, ArrayWrapper1D<char, N> const & rhs ) VULKAN_HPP_NOEXCEPT
      |        ^
/usr/include/vulkan/vulkan_enums.hpp:166:29: note: candidate template ignored: could not match 'const Flags<BitType>' against 'const vk::DeviceMemory'
  166 |   VULKAN_HPP_CONSTEXPR bool operator==( BitType bit, Flags<BitType> const & flags ) VULKAN_HPP_NOEXCEPT
      |                             ^
In file included from <built-in>:1:
In file included from /usr/include/vulkan/vulkan.hpp:8864:
/usr/include/vulkan/vulkan_structs.hpp:7696:75: error: invalid operands to binary expression ('const vk::Buffer' and 'const vk::Buffer')
 7696 |       return ( sType == rhs.sType ) && ( pNext == rhs.pNext ) && ( buffer == rhs.buffer ) && ( memory == rhs.memory ) && ( memoryOffset == rhs.memoryOffset );
      |                                                                    ~~~~~~ ^  ~~~~~~~~~~
/usr/include/vulkan/vulkan.hpp:203:8: note: candidate template ignored: could not match 'const ArrayWrapper1D<char, N>' against 'const vk::Buffer'
  203 |   bool operator==( ArrayWrapper1D<char, N> const & lhs, ArrayWrapper1D<char, M> const & rhs ) VULKAN_HPP_NOEXCEPT
      |        ^
/usr/include/vulkan/vulkan.hpp:248:8: note: candidate template ignored: could not match 'const ArrayWrapper1D<char, N>' against 'const vk::Buffer'
  248 |   bool operator==( std::string const & lhs, ArrayWrapper1D<char, N> const & rhs ) VULKAN_HPP_NOEXCEPT
      |        ^
/usr/include/vulkan/vulkan_enums.hpp:166:29: note: candidate template ignored: could not match 'const Flags<BitType>' against 'const vk::Buffer'
  166 |   VULKAN_HPP_CONSTEXPR bool operator==( BitType bit, Flags<BitType> const & flags ) VULKAN_HPP_NOEXCEPT
      |                             ^
In file included from <built-in>:1:
In file included from /usr/include/vulkan/vulkan.hpp:8864:
/usr/include/vulkan/vulkan_structs.hpp:7696:103: error: invalid operands to binary expression ('const vk::DeviceMemory' and 'const vk::DeviceMemory')
 7696 |       return ( sType == rhs.sType ) && ( pNext == rhs.pNext ) && ( buffer == rhs.buffer ) && ( memory == rhs.memory ) && ( memoryOffset == rhs.memoryOffset );
      |                                                                                                ~~~~~~ ^  ~~~~~~~~~~
/usr/include/vulkan/vulkan.hpp:203:8: note: candidate template ignored: could not match 'const ArrayWrapper1D<char, N>' against 'const vk::DeviceMemory'
  203 |   bool operator==( ArrayWrapper1D<char, N> const & lhs, ArrayWrapper1D<char, M> const & rhs ) VULKAN_HPP_NOEXCEPT
      |        ^
/usr/include/vulkan/vulkan.hpp:248:8: note: candidate template ignored: could not match 'const ArrayWrapper1D<char, N>' against 'const vk::DeviceMemory'
  248 |   bool operator==( std::string const & lhs, ArrayWrapper1D<char, N> const & rhs ) VULKAN_HPP_NOEXCEPT
      |        ^
/usr/include/vulkan/vulkan_enums.hpp:166:29: note: candidate template ignored: could not match 'const Flags<BitType>' against 'const vk::DeviceMemory'
  166 |   VULKAN_HPP_CONSTEXPR bool operator==( BitType bit, Flags<BitType> const & flags ) VULKAN_HPP_NOEXCEPT
      |                             ^
In file included from <built-in>:1:
In file included from /usr/include/vulkan/vulkan.hpp:8864:
/usr/include/vulkan/vulkan_structs.hpp:7818:111: error: invalid operands to binary expression ('const vk::PipelineLayout' and 'const vk::PipelineLayout')
 7818 |       return ( sType == rhs.sType ) && ( pNext == rhs.pNext ) && ( stageFlags == rhs.stageFlags ) && ( layout == rhs.layout ) && ( set == rhs.set );
      |                                                                                                        ~~~~~~ ^  ~~~~~~~~~~
/usr/include/vulkan/vulkan.hpp:203:8: note: candidate template ignored: could not match 'const ArrayWrapper1D<char, N>' against 'const vk::PipelineLayout'
  203 |   bool operator==( ArrayWrapper1D<char, N> const & lhs, ArrayWrapper1D<char, M> const & rhs ) VULKAN_HPP_NOEXCEPT
      |        ^
/usr/include/vulkan/vulkan.hpp:248:8: note: candidate template ignored: could not match 'const ArrayWrapper1D<char, N>' against 'const vk::PipelineLayout'
  248 |   bool operator==( std::string const & lhs, ArrayWrapper1D<char, N> const & rhs ) VULKAN_HPP_NOEXCEPT
      |        ^
/usr/include/vulkan/vulkan_enums.hpp:166:29: note: candidate template ignored: could not match 'const Flags<BitType>' against 'const vk::PipelineLayout'
  166 |   VULKAN_HPP_CONSTEXPR bool operator==( BitType bit, Flags<BitType> const & flags ) VULKAN_HPP_NOEXCEPT
      |                             ^
In file included from <built-in>:1:
In file included from /usr/include/vulkan/vulkan.hpp:8864:
/usr/include/vulkan/vulkan_structs.hpp:8011:111: error: invalid operands to binary expression ('const vk::PipelineLayout' and 'const vk::PipelineLayout')
 8011 |       return ( sType == rhs.sType ) && ( pNext == rhs.pNext ) && ( stageFlags == rhs.stageFlags ) && ( layout == rhs.layout ) && ( firstSet == rhs.firstSet ) &&
      |                                                                                                        ~~~~~~ ^  ~~~~~~~~~~
/usr/include/vulkan/vulkan.hpp:203:8: note: candidate template ignored: could not match 'const ArrayWrapper1D<char, N>' against 'const vk::PipelineLayout'
  203 |   bool operator==( ArrayWrapper1D<char, N> const & lhs, ArrayWrapper1D<char, M> const & rhs ) VULKAN_HPP_NOEXCEPT
      |        ^
/usr/include/vulkan/vulkan.hpp:248:8: note: candidate template ignored: could not match 'const ArrayWrapper1D<char, N>' against 'const vk::PipelineLayout'
  248 |   bool operator==( std::string const & lhs, ArrayWrapper1D<char, N> const & rhs ) VULKAN_HPP_NOEXCEPT
      |        ^
/usr/include/vulkan/vulkan_enums.hpp:166:29: note: candidate template ignored: could not match 'const Flags<BitType>' against 'const vk::PipelineLayout'
  166 |   VULKAN_HPP_CONSTEXPR bool operator==( BitType bit, Flags<BitType> const & flags ) VULKAN_HPP_NOEXCEPT
      |                             ^
In file included from <built-in>:1:
In file included from /usr/include/vulkan/vulkan.hpp:8864:
/usr/include/vulkan/vulkan_structs.hpp:8469:74: error: invalid operands to binary expression ('const vk::Image' and 'const vk::Image')
 8469 |       return ( sType == rhs.sType ) && ( pNext == rhs.pNext ) && ( image == rhs.image ) && ( memory == rhs.memory ) && ( memoryOffset == rhs.memoryOffset );
      |                                                                    ~~~~~ ^  ~~~~~~~~~
/usr/include/vulkan/vulkan.hpp:203:8: note: candidate template ignored: could not match 'const ArrayWrapper1D<char, N>' against 'const vk::Image'
  203 |   bool operator==( ArrayWrapper1D<char, N> const & lhs, ArrayWrapper1D<char, M> const & rhs ) VULKAN_HPP_NOEXCEPT
      |        ^
/usr/include/vulkan/vulkan.hpp:248:8: note: candidate template ignored: could not match 'const ArrayWrapper1D<char, N>' against 'const vk::Image'
  248 |   bool operator==( std::string const & lhs, ArrayWrapper1D<char, N> const & rhs ) VULKAN_HPP_NOEXCEPT
      |        ^
/usr/include/vulkan/vulkan_enums.hpp:166:29: note: candidate template ignored: could not match 'const Flags<BitType>' against 'const vk::Image'
  166 |   VULKAN_HPP_CONSTEXPR bool operator==( Curiously, 06b91cfdc21e is part of `v1.3.302` in Vulkan-Hpp but part of `v1.4.303` in Vulkan-Headers.BitType bit, Flags<BitType> const & flags ) VULKAN_HPP_NOEXCEPT
      |                             ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
```

--- a/VulkanHppGenerator.cpp
+++ b/VulkanHppGenerator.cpp
@@ -228,11 +228,62 @@ void VulkanHppGenerator::generateHandlesHppFile() const
 
 namespace VULKAN_HPP_NAMESPACE
 {
-${structForwardDeclarations}
-${handleForwardDeclarations}
-${uniqueHandles}
-${handles}
+  ${structForwardDeclarations}
+  ${handleForwardDeclarations}
+  ${uniqueHandles}
+  ${handles}
 }   // namespace VULKAN_HPP_NAMESPACE
+
+// operators to compare vk::-handles
+#  if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
+template <typename T, typename std::enable_if<VULKAN_HPP_NAMESPACE::isVulkanHandleType<T>::value, int>::type = 0>
+bool operator<=>( T const & lhs, T const & rhs )
+{
+  return static_cast<typename T::NativeType>( lhs ) <=> static_cast<typename T::NativeType>( rhs );
+}
+#else
+template <typename T, typename std::enable_if<VULKAN_HPP_NAMESPACE::isVulkanHandleType<T>::value, int>::type = 0>
+bool operator==( T const & lhs, T const & rhs )
+{
+  return static_cast<typename T::NativeType>( lhs ) == static_cast<typename T::NativeType>( rhs );
+}
+
+template <typename T, typename std::enable_if<VULKAN_HPP_NAMESPACE::isVulkanHandleType<T>::value, int>::type = 0>
+bool operator!=( T const & lhs, T const & rhs )
+{
+  return static_cast<typename T::NativeType>( lhs ) != static_cast<typename T::NativeType>( rhs );
+}
+
+template <typename T, typename std::enable_if<VULKAN_HPP_NAMESPACE::isVulkanHandleType<T>::value, int>::type = 0>
+bool operator<( T const & lhs, T const & rhs )
+{
+  return static_cast<typename T::NativeType>( lhs ) < static_cast<typename T::NativeType>( rhs );
+}
+#endif
+
+template <typename T, typename std::enable_if<VULKAN_HPP_NAMESPACE::isVulkanHandleType<T>::value, int>::type = 0>
+bool operator==( T const & v, std::nullptr_t )
+{
+  return !v;
+}
+
+template <typename T, typename std::enable_if<VULKAN_HPP_NAMESPACE::isVulkanHandleType<T>::value, int>::type = 0>
+bool operator==( std::nullptr_t, T const & v )
+{
+  return !v;
+}
+
+template <typename T, typename std::enable_if<VULKAN_HPP_NAMESPACE::isVulkanHandleType<T>::value, int>::type = 0>
+bool operator!=( T const & v, std::nullptr_t )
+{
+  return !!v;
+}
+
+template <typename T, typename std::enable_if<VULKAN_HPP_NAMESPACE::isVulkanHandleType<T>::value, int>::type = 0>
+bool operator!=( std::nullptr_t, T const & v )
+{
+  return !!v;
+}
 #endif
 )";
 
@@ -8018,7 +8069,7 @@ ${typesafeConversionConditionalEnd}
       return *this;
     }
 
-${commands}
+    ${commands}
     ${typesafeExplicitKeyword}operator Vk${className}() const VULKAN_HPP_NOEXCEPT
     {
       return m_${memberName};

--- a/vulkan/vulkan_handles.hpp
+++ b/vulkan/vulkan_handles.hpp
@@ -17707,4 +17707,55 @@ namespace VULKAN_HPP_NAMESPACE
 #endif /* VULKAN_HPP_DISABLE_ENHANCED_MODE */
 
 }  // namespace VULKAN_HPP_NAMESPACE
+
+// operators to compare vk::-handles
+#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
+template <typename T, typename std::enable_if<VULKAN_HPP_NAMESPACE::isVulkanHandleType<T>::value, int>::type = 0>
+bool operator<=>( T const & lhs, T const & rhs )
+{
+  return static_cast<typename T::NativeType>( lhs ) <=> static_cast<typename T::NativeType>( rhs );
+}
+#else
+template <typename T, typename std::enable_if<VULKAN_HPP_NAMESPACE::isVulkanHandleType<T>::value, int>::type = 0>
+bool operator==( T const & lhs, T const & rhs )
+{
+  return static_cast<typename T::NativeType>( lhs ) == static_cast<typename T::NativeType>( rhs );
+}
+
+template <typename T, typename std::enable_if<VULKAN_HPP_NAMESPACE::isVulkanHandleType<T>::value, int>::type = 0>
+bool operator!=( T const & lhs, T const & rhs )
+{
+  return static_cast<typename T::NativeType>( lhs ) != static_cast<typename T::NativeType>( rhs );
+}
+
+template <typename T, typename std::enable_if<VULKAN_HPP_NAMESPACE::isVulkanHandleType<T>::value, int>::type = 0>
+bool operator<( T const & lhs, T const & rhs )
+{
+  return static_cast<typename T::NativeType>( lhs ) < static_cast<typename T::NativeType>( rhs );
+}
+#endif
+
+template <typename T, typename std::enable_if<VULKAN_HPP_NAMESPACE::isVulkanHandleType<T>::value, int>::type = 0>
+bool operator==( T const & v, std::nullptr_t )
+{
+  return !v;
+}
+
+template <typename T, typename std::enable_if<VULKAN_HPP_NAMESPACE::isVulkanHandleType<T>::value, int>::type = 0>
+bool operator==( std::nullptr_t, T const & v )
+{
+  return !v;
+}
+
+template <typename T, typename std::enable_if<VULKAN_HPP_NAMESPACE::isVulkanHandleType<T>::value, int>::type = 0>
+bool operator!=( T const & v, std::nullptr_t )
+{
+  return !!v;
+}
+
+template <typename T, typename std::enable_if<VULKAN_HPP_NAMESPACE::isVulkanHandleType<T>::value, int>::type = 0>
+bool operator!=( std::nullptr_t, T const & v )
+{
+  return !!v;
+}
 #endif

